src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -342,11 +341,8 @@ public function nack(\AMQPEnvelope $message, string $queueName, int $flags = AMQ
 
     public function setup(): void
     {
-        if (!$this->channel()->isConnected()) {
-            $this->clear();
-        }
-
         $this->exchange()->declareExchange();
+        $this->getDelayExchange()->declareExchange();
```

Comment:
setup is called from messenger:setup-transports which should also create the delay exchange. this way, auto_setup can actually be disabled successfully even when using delays/retries